gabbi/runner.py

```diff
@@ -88,27 +88,59 @@ def run():
         help='Custom response handler. Should be an import path of the '
              'form package.module or package.module:class.'
     )
+    parser.add_argument(
+        '-f',
+        nargs='?', default=None,
+        dest='input_files',
+        action='append',
+        help='input files'
+    )
 
     args = parser.parse_args()
     host, port, prefix, force_ssl = utils.host_info_from_target(
         args.target, args.prefix)
 
-    # Initialize response handlers.
-    handler_objects = initialize_handlers(args.response_handlers)
+    response_handlers = initialize_handlers(args.response_handlers)
+
+    input_files = args.input_files
+    if input_files is None:
+        input_files = [sys.stdin]
+
+    failfast = args.failfast
+    failure = False
+    for input_file in input_files:
+        params = (response_handlers, host, port, prefix, force_ssl, failfast)
+        # XXX(FND): special-casing; use generic stream detection instead?
```

Comment:
Rather than special casing here, another option would be to assemble the arguments before checking files and then do something like:

```
if input_file is None:
    execute(...)
else:
    for input_file in input_files:
        executre(...)